backend/src/db-utils.js

```diff
@@ -962,7 +962,9 @@ const queryBlocksList = async (limit = 15, paginationIndexer) => {
       ORDER BY blocks.block_timestamp DESC`,
       {
         limit,
-        pagination_indexer: paginationIndexer,
+        pagination_indexer: paginationIndexer
+          ? new BN(paginationIndexer).muln(10 ** 6).toString()
```

Comment:
Ah, I see it now, we return `DIV(blocks.block_timestamp, 1000*1000)`, and then we messed up to multiply it back. This logic is not intuitive at all, but let's merge it as is and address it some time in the future.